### ISSUE TITLE
Exception error is thrown when deleting a card detail

### ISSUE THREAD

**Issue by ednamaeG** [wekan/wekan#3252]
**Server Setup Information:::**

Did you test in newest Wekan?: Yes
For new Wekan install, did you configure root-url correctly https://github.com/wekan/wekan/wiki/Settings ? Yes
Wekan version: 4.09
Operating System: Ubuntu 18.04
Deployment Method(snap/docker/sandstorm/mongodb bundle/source): Source?
Http frontend if any (Caddy, Nginx, Apache, see config examples from Wekan GitHub wiki first): Nginx
Node Version: v12.16.1
MongoDB Version: 4.2.5

Wekan only works on newest desktop Firefox/Chromium/Chrome/Edge/Chromium Edge and mobile Chrome. What webbrowser version are you using? Chrome

**Problem description:**
- Removed nonexistent document error is thrown when deleting a card detail which causes the server to load too slow. 
- Bug also exists in the latest version. 
- Related to #3203 , #1629
 
**Steps to Reproduce**
-  **Archive** or **Add** 4 or more cards consecutively / select all cards in a specific list then click **Move Selection to Archive** on the sidebar
-  **Delete** something from the card details (comment or checklist) .

Here is the error: 
 (this was tested from a local development server)

![bandicam-2020-09-02-08-41-43-481](https://user-images.githubusercontent.com/49261930/91920753-8675c400-ecfc-11ea-9d0c-4eaba99d3661.gif)

![image](https://user-images.githubusercontent.com/49261930/91921349-def99100-ecfd-11ea-9f02-ac1a031b8d48.png)

**Comment by xet7** [comment]
@ednamaeG

Does this still happen in newest Wekan?

**Comment by ednamaeG** [comment]
@xet7 yes, it still happens when you try to invite a user to a board then archive a card after inviting.

**Comment by xator91** [comment]
It still happen. And when you delete a lot of card and got a lot of errors, cpu usage is at maximum and need to restart.

**Comment by xator91** [comment]
From 2020 still existing this bug related to #5322 @xet7

**Comment by xet7** [comment]
Partial fix in the upcoming WeKan release; leaving this open because one part remains.

**What was fixed:** the thrown `Removed nonexistent document` exception when deleting a comment or checklist/checklist-item. The client delete handlers called `Collection.remove(_id)` directly, but under heavy archive/delete churn the target doc can already have been evicted from the browser's Minimongo cache — and removing a missing `_id` throws. The comment / checklist / checklist-item delete handlers now check the doc still exists in the local cache before removing it (`client/components/activities/comments.js`, `client/components/cards/checklists.js`). The server-side `before.remove` hooks were already hardened earlier to guard + log instead of throwing.

**What still remains (why this stays open):** the **high CPU on bulk delete** that @xator91 reported (and the relation to #5322) is a separate issue — it's reactivity/mergebox churn from the delete cascade (`cardRemover` fires hooked removes for every child doc, each generating activities/notifications). That's a larger performance effort, not a single throw site, so it's tracked separately.

Note: there's no automated regression test for the guard — the eviction race isn't deterministically reproducible — but the guard itself is a self-evident findOne-then-remove. If you can still reproduce the *thrown exception* (not the CPU) on the next release, please attach the exact console stack and steps.
